pages/blog/asyncapi-say-hi-to-triggermesh.md

```diff
@@ -0,0 +1,272 @@
+/>
+
+The project includes a [prototype parser](https://github.com/triggermesh/tutorials/blob/main/asyncAPI/parser.js) built with AsyncAPI's [js-parser](https://github.com/asyncapi/parser-js). It parses the [provided AsyncAPI definition](https://github.com/triggermesh/tutorials/blob/main/asyncAPI/asyncapi.yaml) and generates `tmctl` CLI commands that create TriggerMesh event sources that will ingest events from the various channels into a TriggerMesh broker for further processing. You could of course write the `tmctl` commands manually too, but this parser shows how a well-documented AsyncAPI definition can be used alongside TriggerMesh.
+
+The following schema illustrates what we'll build in this tutorial.
+
+<Figure
+  src="/img/posts/asyncapi-say-hi-to-triggermesh/full-schema.webp"
+  caption="Schema shows the event flow we'll build during the tutorial"
+/>
+
+Please reach out to the TriggerMesh team on [Slack](https://join.slack.com/t/triggermesh-community/shared_invite/zt-1kngevosm-MY7kqn9h6bT08hWh8PeltA) or [GitHub](https://github.com/triggermesh/tutorials/issues/new) if you need help getting this to work or have any feedback.
+
+## Prerequisites
+
+### `tmctl`
+
+Install `tmctl` with homebrew or [other methods](https://docs.triggermesh.io/get-started/quickstart/):
+
+```sh
+brew install triggermesh/cli/tmctl
+```
+
+### AsyncAPI's `js-parser`
+
+We also need to install the AsyncAPI parser. Make sure you have a recent enough version of npm and node:
+
+```sh
+npm install @asyncapi/parser
+```
+
+### Quickstart a Kafka cluster
+
+One of the order management systems is producing `orders` on a Kafka topic called `orders`. We also want to write all unified orders to a downstream `unified-orders` topic. We'll need a cluster to implement these topics.
+
+Here we provide an easy way to start one using [Redpanda](https://redpanda.com/) and Docker Compose, but you could use any Kafka distribution you like (self-hosted or managed).
+
+The provided [docker-compose file](https://github.com/triggermesh/tutorials/blob/main/asyncAPI/docker-compose.yml) will start a single node Redpanda cluster and a handy console. It is configured to work with docker desktop and could require some adjustments to the listeners and advertised listeners for it to work on other environments. Reach out to us on Slack or GitHub if you need help, or see [here](https://www.confluent.io/blog/kafka-listeners-explained/) if you want to deep dive on this.
+
+Run the following command in the same directory as the docker compose file to start Redpanda:
+
+```sh
+docker-compose up -d
+```
+
+### Start a mock HTTP service
+
+We'll start a mock HTTP service locally to simulate the order management system that provides `orders` through an HTTP API. Copy the [order.json](https://github.com/triggermesh/tutorials/blob/main/asyncAPI/order.json) file to the current directory, it we be used as a mock event. Run the mock HTTP server, ideally in a new terminal (requires python 2 or 3):
```

Comment:
```suggestion
We'll start a mock HTTP service locally to simulate the order management system that provides `orders` through an HTTP API. Copy the [order.json](https://github.com/triggermesh/tutorials/blob/main/asyncAPI/order.json) file to the current directory, it will be used as a mock event. Run the mock HTTP server, ideally in a new terminal (requires python 2 or 3):
```